### PR TITLE
OSS-205 - email-queue

### PR DESCRIPTION
**Pull request by zsigmond-czine-everit**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  <groupId>org.everit.email</groupId>
+  <artifactId>org.everit.email.queue</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+
+  <packaging>bundle</packaging>
+
+  <name>Everit - Email Queue</name>
+
+  <properties>
+    <projectpath>email-store-ri</projectpath>
```
Comment (balazs-zsoldos) [inline core/pom.xml]: email-queue I guess

File: core/src/main/java/org/everit/email/queue/CreatePassOnJobParam.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package org.everit.email.queue;
+
+/**
+ * Information to create pass on job.
+ */
+public class CreatePassOnJobParam {
+
+  public int max;
+
+  public CreatePassOnJobParam max(final int max) {
```
Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/CreatePassOnJobParam.java]: function name should start with small case letter

Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/CreatePassOnJobParam.java]: Function name should start with small-case letter

File: core/src/main/java/org/everit/email/queue/CreatePassOnJobParam.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.everit.email.queue;
+
+/**
+ * Information to create pass on job.
+ */
+public class CreatePassOnJobParam {
```
Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/CreatePassOnJobParam.java]: We do not need Create, just PassOnJobParam

File: core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+   * Queue email data holder.
+   */
+  public static class QueueEmailDTO {
+
+    public long queueEmailId;
+
+    public long storedEmailId;
+  }
+
+  private EmailSender emailSender;
```
Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java]: Let's call the variable "sink". Also call it in the parameters of the constructors "sink"

File: core/src/main/java/org/everit/email/queue/EmailQueue.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+   * @param emailSender
+   *          an {@link EmailSender} instance. Cannot be <code>null</code>!
+   * @param emailStore
+   *          an {@link EmailStore} instance. Cannot be <code>null</code>!
+   * @param querydslSupport
+   *          a {@link QuerydslSupport} instance. Cannot be <code>null</code>!
+   * @param transactionPropagator
+   *          a {@link TransactionPropagator} instance. Cannot be <code>null</code>!
+   */
+  public EmailQueue(final EmailSender emailSender, final EmailStore emailStore,
```
Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/EmailQueue.java]: Create a paremter class called EmailQueueParameter. Later we can have more configuration like "parallelPassOn" if skip locked will be supported by QueryDSL.

File: core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    // do nothing
+  }
+
+  /**
+   * Creates pass on job that send mails.
+   */
+  public Runnable createPassOnJob(final CreatePassOnJobParam param) {
+    return () -> {
+      transactionPropagator.required(() -> {
+        List<QueueEmailDTO> queuedEmails = selectAndLockQueueEmail(param);
```
Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java]: I would call QueuedEmailDTO and everywhere else I would take there the "d" letter

File: core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+ * Email Queue Bulk email sender implementation.
+ */
+public class EmailQueueBulkEmailSender implements BulkEmailSender {
+
+  /**
+   * Queue email data holder.
+   */
+  public static class QueueEmailDTO {
+
+    public long queueEmailId;
```
Comment (balazs-zsoldos) [inline core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java]: queuedEmailId

**Comment by zsigmond-czine-everit** [comment]
@balazs-zsoldos:
Code updated.

**Pull request merged**